reconcile/utils/deadmanssnitch_api.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from typing import (
+    Any,
+    Optional,
+    Self,
+)
+
+import requests
+from pydantic import BaseModel
+
+BASE_URL = "https://api.deadmanssnitch.com/v1/snitches"
+
+
+class DeadManssnitchException(Exception):
+    pass
+
+
+class Snitch(BaseModel):
+    token: str
+    href: str
+    name: str
+    tags: list[str]
+    notes: str
+    status: str
+    check_in_url: str
+    interval: str
+    alert_type: str
+    alert_email: list[str]
+    vault_data: Optional[str]
+
+    def get_cluster_name(self) -> str:
+        return self.name.split(".")[1]
+
+    def to_dict(self) -> dict[str, Any]:
+        return self.dict(by_alias=True)
+
+
+class DeadMansSnitchApi:
+    def __init__(self, token: str, url: str = BASE_URL) -> None:
+        self.token = token
+        self.url = url
+        self.session = requests.Session()
+
+    def __enter__(self) -> Self:
+        return self
+
+    def __exit__(self, exc_type: Any, exc_value: Any, traceback: Any) -> None:
+        self.session.close()
+
+    def get_snitches(self, tags: list[str]) -> list[Snitch]:
+        full_url = f"{self.url}?tags={','.join(tags)}"
+        logging.debug("Getting snitches for tags:%s", tags)
+        response = self.session.get(url=full_url, auth=(self.token, ""))
+        response.raise_for_status()
+        snitches = [Snitch(**item) for item in response.json()]
+        return snitches
+
+    def create_snitch(self, payload: dict) -> Snitch:
```

Comment:
Can you use a BaseModel for the payload?

Reply:
@janboll add baseModel would be enforcing all the fields here...https://github.com/app-sre/qontract-reconcile/blob/f0c369d5c391d1db90ef56a97588b076ca7bae3b/reconcile/deadmanssnitch.py#L90to99 ..the fact is deadmanssnith does not require all these fields...this file is supposed to be close to the actual api.deadmanssnitch.com in its behavior hence a dict was chosen